lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -3123,11 +3123,19 @@ TypeAndOrName SwiftLanguageRuntimeImpl::FixUpDynamicType(
 
 bool SwiftLanguageRuntimeImpl::IsTaggedPointer(lldb::addr_t addr,
                                                CompilerType type) {
-  swift::CanType swift_can_type = GetCanonicalSwiftType(type);
-  if (!swift_can_type)
+  Demangler dem;
+  auto *node = dem.demangleSymbol(type.GetMangledTypeName().GetStringRef());
+  Node::Kind kind_path[] = {Node::Kind::Global, Node::Kind::TypeMangling,
+                            Node::Kind::Type};
+  for (auto kind : kind_path) {
+    if (node && node->getKind() == kind && node->hasChildren()) {
+      node = node->getFirstChild();
+      continue;
```

Comment:
Uh this is neat! Maybe we should factor this out into a path matcher helper?

Reply:
In this PR or a separate one?

Reply:
I'll do it in stable, not the release branch.

Reply:
If you're actually going to do it, we'll probably still want to cherry-pick it, because it will mess with our ability to cherry-pick subsequent patches.

Reply:
@adrian-prantl I added a helper function, `nodeAtPath`, in a new file `SwiftDemangle.h`.